packages/react-native-platform-override/src/Hash.ts

```diff
@@ -43,9 +44,29 @@ export class Hasher {
  * Convenience helper which hashes a single buffer
  */
 export function hashContent(content: string | Buffer): string {
-  if (typeof content === 'string') {
-    content = Buffer.from(content);
+  return new Hasher().feedContent(content).digest();
+}
+
+/**
+ * Helper to hash a file/directory belonging to a repository
+ */
+export async function hashFileOrDirectory(
+  name: string,
+  repo: FileRepository,
+): Promise<string | null> {
+  const type = await repo.stat(name);
+  if (type === 'none') {
+    return null;
   }
 
-  return new Hasher().feedContent(content).digest();
+  if (type === 'file') {
+    return hashContent((await repo.readFile(name))!);
+  } else {
+    const hasher = new Hasher();
+    for (const file of (await repo.listFiles([`${name}/**`])).sort()) {
+      const contents = await repo.readFile(file);
+      hasher.feedContent(contents!);
```

Comment:
>hasher [](start = 6, length = 6)

I would recommend also hashing the filenames to catch renames that don't affect the sort-order.

Reply:
That's a good point. Will add that.

Reply:
I just realize we don't actually remove old files either. Need to add that as well.